Mu2eKinKal/src/LoopHelixFit_module.cc

```diff
@@ -0,0 +1,558 @@
+      ComboHitCollection const& chcol, StrawHitIndexCollection const& strawHitIdxs,
+      MEASCOL& thits,EXINGCOL& exings) const {
+  // wrap the seed traj in a Piecewise traj: needed to satisfy PTOCA interface
+    PKTRAJ pseedtraj(seedtraj);
+    // loop over the individual straw hits
+    for(auto strawidx : strawHitIdxs) {
+      const ComboHit& strawhit(chcol.at(strawidx));
+      if(strawhit.mask().level() != StrawIdMask::uniquestraw)
+	throw cet::exception("RECO")<<"mu2e::LoopHelixFit: ComboHit error"<< endl;
+      const Straw& straw = tracker.getStraw(strawhit.strawId());
+      // find the propagation speed for signals along this straw
+      double sprop = 2*strawresponse.halfPropV(strawhit.strawId(),strawhit.energyDep());
+      // construct a kinematic line trajectory from this straw. the measurement point is the signal end
+      auto p0 = straw.wireEnd(strawhit.driftEnd());
+      auto p1 = straw.wireEnd(StrawEnd(strawhit.driftEnd().otherEnd()));
+      auto propdir = (p0 - p1).unit(); // The signal propagates from the far end to the near
+      // clumsy conversion: make this more elegant TODO
+      VEC3 vp0(p0.x(),p0.y(),p0.z());
+      VEC3 vp1(p1.x(),p1.y(),p1.z());
+      Line wline(vp0,vp1,strawhit.time(),sprop);
+      // compute 'hint' to TPOCA.  correct the hit time using the time division
+      double psign = propdir.dot(straw.wireDirection());  // wire distance is WRT straw center, in the nominal wire direction
+      double htime = wline.t0() - (straw.halfLength()-psign*strawhit.wireDist())/wline.speed();
+      CAHint hint(seedtraj.ztime(vp0.Z()),htime);
+      // compute a preliminary PTCA between the seed trajectory and this straw.
+      PTCA ptca(pseedtraj, wline, hint, tpocaprec_);
+      // create the material crossing
+      if(addmat_){
+	exings.push_back(std::make_shared<STRAWXING>(ptca,*smat_));
+	if(print_ > 2)exings.back()->print(std::cout,1);
+      }
+      // create the hit
+      thits.push_back(std::make_shared<KKSTRAWHIT>(*kkbf_, ptca, whstate_, strawhit, straw, strawidx, strawresponse));
+      // 
+      if(print_ > 2)thits.back()->print(std::cout,2);
+    }
+  }
+
+  void LoopHelixFit::makeCaloHit(CCPtr const& cluster, MEASCOL& thits) const {
+    // move cluster COG into the tracker frame.  COG is at the front face of the disk
+    GeomHandle<mu2e::Calorimeter> calo_h;
+    CLHEP::Hep3Vector cog = calo_h->geomUtil().mu2eToTracker(calo_h->geomUtil().diskFFToMu2e( cluster->diskID(), cluster->cog3Vector()));
+    // project this along the crystal axis to the SIPM, which is at the back.  This is the point the time measurement corresponds to
+    VEC3 FFCOG(cog);
+    VEC3 SIPMCOG = FFCOG + crystalF2B_;
+    // create the Line trajectory from this information: signal goes towards the sipm
+    Line cogaxis(SIPMCOG,FFCOG,cluster->time()+caloDt_,caloPropSpeed_); 
+    // create the hit
+    double tvar = cluster->timeErr()*cluster->timeErr();
```

Comment:
The geometry description is correct, but the timing error might still be too optimistic and that might slightly skew the results. I'll let you experiment.